src/message-authentication/hash-based.md

```diff
@@ -0,0 +1,63 @@
+# Hash-Based Message Authentication (HMAC)
+
+HMACs are [MACs](index.html) based on a hash algorithm and are named after that algorithm; e.g.
+`HMAC-SHA256`. Usually, a HMAC algorithm will concatenate the data and the secret then hash them
+together to create the signature.
```

Comment:
Do you want to say that the content comes first so that pre-computing isn't feasible?  Probably not, just thought I'd mention it.

Reply:
I tried to keep the sections pretty light and beginner friendly - but I was planning to add a `## Detailed Explanation` section in a follow up.

Any change you want to write that section? 😆

Reply:
Doing a detailed section might make sense, but there's the risk of it just being a collection of random facts. So what you have is fine, and it's good to have a consistent audience level!